src/Access/EnabledRowPolicies.h

```diff
@@ -39,8 +52,8 @@ class EnabledRowPolicies
     /// Returns prepared filter for a specific table and operations.
     /// The function can return nullptr, that means there is no filters applied.
     /// The returned filter can be a combination of the filters defined by multiple row policies.
-    ASTPtr getFilter(const String & database, const String & table_name, RowPolicyFilterType filter_type) const;
-    ASTPtr getFilter(const String & database, const String & table_name, RowPolicyFilterType filter_type, const ASTPtr & combine_with_expr) const;
+    RowPolicyFilter getFilter(const String & database, const String & table_name, RowPolicyFilterType filter_type) const;
+    RowPolicyFilter getFilter(const String & database, const String & table_name, RowPolicyFilterType filter_type, const RowPolicyFilter & combine_with_filter) const;
```

Comment:
`std::shared_ptr<const RowPolicyFilter>` as a result type seems a bit better to me. Because this way we won't need to copy that vector if we store `std::shared_ptr<const RowPolicyFilter>` inside `MixedFiltersResult`. And if there is no filter the function will return just nullptr.

Reply:
Done.